### PR TITLE
chore: enable VHD 2019.05.08, disable auditd for non-VHD

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Add new VHD, and disable auditd (which is introduced via VHD in 2019.05.08) for non-VHD distros

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -397,16 +397,16 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 		})
 
 		It("should validate auditd configuration", func() {
-			if eng.ExpandedDefinition.Properties.IsUbuntuDistroForAllNodes() {
-				var enabledProfiles []string
+			if eng.ExpandedDefinition.Properties.IsVHDDistroForAllNodes() {
```
Comment (jackfrancis) [inline test/e2e/kubernetes/kubernetes_test.go]: Now only running these tests in VHD scenario

File: test/e2e/kubernetes/node/node.go [hunk 2]
```diff
@@ -108,10 +108,10 @@ func (n *Node) IsUbuntu() bool {
 	return false
 }
 
-// IsInProfile determines if this node is running on a vm with auditd enabled
-func (n *Node) IsInProfile(profiles []string) bool {
-	for _, profile := range profiles {
-		if strings.Contains(strings.ToLower(n.Metadata.Name), profile) {
+// HasSubstring determines if a node name matches includes the passed in substring
```
Comment (jackfrancis) [inline test/e2e/kubernetes/node/node.go]: Renamed this helper method to more generally express what it actually does

File: parts/k8s/cloud-init/masternodecustomdata.yml [hunk 3]
```diff
@@ -104,15 +104,6 @@ write_files:
   content: !!binary |
     {{CloudInitData "aptPreferences"}}
 
-{{if .MasterProfile.IsUbuntuNonVHD}}
```
Comment (CecileRobertMichon) [inline parts/k8s/cloud-init/masternodecustomdata.yml]: should we keep the if/end as scaffolding to facilitate development for future CIS files that get added to the Ubuntu distro first?

File: pkg/engine/armvariables.go [hunk 4]
```diff
@@ -149,7 +149,6 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		masterVars["cloudInitFiles"].(map[string]interface{})["profileDCISSh"] = getBase64EncodedGzippedCustomScript(profileDCISSh)
 		masterVars["cloudInitFiles"].(map[string]interface{})["pamDotDCommonAuth"] = getBase64EncodedGzippedCustomScript(pamDotDCommonAuth)
 		masterVars["cloudInitFiles"].(map[string]interface{})["pamDotDCommonPassword"] = getBase64EncodedGzippedCustomScript(pamDotDCommonPassword)
```
Comment (CecileRobertMichon) [inline pkg/engine/armvariables.go]: I forgot to remove these in the other PR, they can probably all go

File: pkg/engine/armvariables_test.go [hunk 5]
```diff
@@ -222,7 +222,6 @@ func TestK8sVars(t *testing.T) {
 		"profileDCISSh":                    getBase64EncodedGzippedCustomScript(profileDCISSh),
 		"pamDotDCommonAuth":                getBase64EncodedGzippedCustomScript(pamDotDCommonAuth),
 		"pamDotDCommonPassword":            getBase64EncodedGzippedCustomScript(pamDotDCommonPassword),
```
Comment (CecileRobertMichon) [inline pkg/engine/armvariables_test.go]: same here

File: pkg/engine/const.go [hunk 6]
```diff
@@ -180,7 +180,6 @@ const (
 	profileDCISSh                            = "k8s/cloud-init/artifacts/profile-d-cis.sh"
 	pamDotDCommonAuth                        = "k8s/cloud-init/artifacts/pam-d-common-auth"
 	pamDotDCommonPassword                    = "k8s/cloud-init/artifacts/pam-d-common-password"
```
Comment (CecileRobertMichon) [inline pkg/engine/const.go]: and here

File: pkg/api/azenvtypes.go [hunk 7]
```diff
@@ -137,15 +137,15 @@ var (
 		ImageOffer:     "aks",
 		ImageSku:       "aks-ubuntu-1604-201904",
```
Comment (CecileRobertMichon) [inline pkg/api/azenvtypes.go]: SKU needs to be updated

**Pull request merged**